buildSrc/src/main/java/org/opensearch/gradle/agent/JavaAgent.java

```diff
@@ -0,0 +1,85 @@
+ * compatible open source license.
+ */
+
+package org.opensearch.gradle.agent;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.tasks.Copy;
+import org.gradle.api.tasks.TaskProvider;
+import org.gradle.api.tasks.testing.Test;
+
+import java.io.File;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Gradle plugin to automatically configure the OpenSearch Java agent
+ * for test tasks in OpenSearch plugin projects.
+ */
+public class JavaAgent implements Plugin<Project> {
+
+    /**
+     * Plugin implementation that sets up java agent configuration and applies it to test tasks.
+     */
+    @Override
+    public void apply(Project project) {
+        Configuration agentConfiguration = project.getConfigurations().findByName("agent");
+        if (agentConfiguration == null) {
+            agentConfiguration = project.getConfigurations().create("agent");
+        }
+
+        project.afterEvaluate(p -> {
+            String opensearchVersion = getOpensearchVersion(p);
+            String byteBuddyVersion = getByteBuddyVersion(p);
+
+            p.getDependencies().add("agent", "org.opensearch:opensearch-agent-bootstrap:" + opensearchVersion);
+            p.getDependencies().add("agent", "org.opensearch:opensearch-agent:" + opensearchVersion);
+            p.getDependencies().add("agent", "net.bytebuddy:byte-buddy:" + byteBuddyVersion);
+        });
+
+        Configuration finalAgentConfiguration = agentConfiguration;
+        TaskProvider<Copy> prepareJavaAgent = project.getTasks().register("prepareJavaAgent", Copy.class, task -> {
+            task.from(finalAgentConfiguration);
+            task.into(new File(project.getBuildDir(), "agent"));
+        });
+
+        project.getTasks().withType(Test.class).configureEach(testTask -> {
+            testTask.dependsOn(prepareJavaAgent);
+
+            final String opensearchVersion = getOpensearchVersion(project);
+
+            testTask.doFirst(task -> {
+                File agentJar = new File(project.getBuildDir(), "agent/opensearch-agent-" + opensearchVersion + ".jar");
+
+                testTask.jvmArgs("-javaagent:" + agentJar.getAbsolutePath());
+            });
+        });
+    }
+
+    /**
+     * Gets the OpenSearch version from project properties, with a fallback default.
+     *
+     * @param project The Gradle project
+     * @return The OpenSearch version to use
+     */
+    private String getOpensearchVersion(Project project) {
+        return Objects.requireNonNull(project.property("opensearch_version")).toString();
+    }
+
+    /**
+     * Gets the ByteBuddy version from project properties, with a fallback default.
+     *
+     * @param project The Gradle project
+     * @return The ByteBuddy version to use
+     */
+    private String getByteBuddyVersion(Project project) {
+        return (String) ((Map<?, ?>) project.property("versions")).get("bytebuddy");
```

Comment:
nit: Make a generic `getVersion` and pass bytebuddy as an arg.

Is it possible to do one cast here for `Map<String, String>`?

Reply:
let me see, I might end up in unchecked cast warnings but I can always use `@SuppressWarnings("unchecked")`

Reply:
Updated @cwperks

Reply:
But thinking back is the original logic `return (String) ((Map<?, ?>) project.property("versions")).get("bytebuddy");` better ? as it only casts the bytebuddy which is just what we need.